### PR TITLE
Fake schema data oliver

### PR DESCRIPTION
**Pull request by olliejth**
Changed the fake data upload sql file after change of architecture diagram and ERD. Data is not verified with correct information but stands as a potential example of what it might look like.

### CODE REVIEW

File: database/fake_data.sql [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+('2024-09-30 14:20:35', 2),
+('2024-10-01 08:45:20', 1),
+('2024-10-02 19:30:50', 3),
+('2024-10-03 11:55:15', 1),
+('2024-10-04 16:10:40', 2),
+('2024-10-05 09:25:30', 1),
+('2024-10-06 13:50:45', 3),
+('2024-10-06 18:05:55', 1),
+('2024-10-07 07:40:10', 2),
+('2024-10-07 22:15:25', 1);
```
Comment (CalebMathers) [inline database/fake_data.sql:29]: Good format, won't be any issues with the tables

File: database/fake_data.sql [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+(12, 18.5, 30, 1.1, 20, 10000, '2024-09-30 14:20:35'),
+(45, 15.2, 60, 2.5, 85, 8000, '2024-10-01 09:30:45'),
+(8, 21.1, 10, 1.1, 10, 15000, '2024-10-02 18:15:20'),
+(91, 13.8, 70, 3.2, 90, 5000, '2024-10-03 12:45:55'),
+(67, 17.6, 40, 1.5, 60, 12000, '2024-10-04 16:25:10'),
+(23, 20.1, 20, 1.1, 25, 14000, '2024-10-05 08:50:05'),
+(58, 14.3, 80, 4.8, 95, 6000, '2024-10-06 14:55:35'),
+(30, 19.4, 50, 1.2, 45, 11000, '2024-10-06 19:20:50'),
+(7, 22.1, 15, 1.1, 12, 16000, '2024-10-07 07:10:25'),
+(83, 16.5, 65, 2.9, 80, 9000, '2024-10-07 22:35:40');
```
Comment (CalebMathers) [inline database/fake_data.sql:43]: Lovely bit of forecast data

File: database/fake_data.sql [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+('chrislee', '+12345678909', 'chrislee@example.com'),
+('lauragarcia', '+12345678910', 'lauragarcia@example.com')
+('robertmartin', '+12345678911', 'robertmartin@example.com'),
+('lisawalker', '+12345678912', 'lisawalker@example.com'),
+('tommyadams', '+12345678913', 'tommyadams@example.com'),
+('nataliebell', '+12345678914', 'nataliebell@example.com'),
+('jacobwhite', '+12345678915', 'jacobwhite@example.com'),
+('oliviayoung', '+12345678916', 'oliviayoung@example.com'),
+('ethanturner', '+12345678917', 'ethanturner@example.com'),
+('gracehall', '+12345678918', 'gracehall@example.com'),
```
Comment (CalebMathers) [inline database/fake_data.sql:98]: solid emails and numbers

File: database/fake_data.sql [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+('ASTRO_star', 'https://dq0hsqwjhea1.cloudfront.net/Interactive-Sky-Chart-600px.jpg', '2024-10-06', 6, 6),
+('ASTRO_star', 'https://dq0hsqwjhea1.cloudfront.net/Interactive-Sky-Chart-600px.jpg', '2024-10-07', 7, 7),
+('NASA_apod', 'https://science.nasa.gov/wp-content/uploads/2023/09/Carina_Nebula-1.jpeg', '2024-10-08', 8, 8),
+('NASA_apod', 'https://science.nasa.gov/wp-content/uploads/2023/09/Carina_Nebula-1.jpeg', '2024-10-09', 9, 9),
+('ASTRO_moon', 'https://upload.wikimedia.org/wikipedia/commons/thumb/e/e1/FullMoon2010.jpg/1200px-FullMoon2010.jpg', '2024-10-10', 10, 10),
+('ASTRO_star', 'https://dq0hsqwjhea1.cloudfront.net/Interactive-Sky-Chart-600px.jpg', '2024-10-11', 11, 11),
+('NASA_apod', 'https://science.nasa.gov/wp-content/uploads/2023/09/Carina_Nebula-1.jpeg', '2024-10-12', 12, 12),
+('ASTRO_moon', 'https://upload.wikimedia.org/wikipedia/commons/thumb/e/e1/FullMoon2010.jpg/1200px-FullMoon2010.jpg', '2024-10-13', 13, 13),
+('ASTRO_star', 'https://dq0hsqwjhea1.cloudfront.net/Interactive-Sky-Chart-600px.jpg', '2024-10-14', 14, 14),
+('NASA_apod', 'https://science.nasa.gov/wp-content/uploads/2023/09/Carina_Nebula-1.jpeg', '2024-10-15', 15, 15),
```
Comment (CalebMathers) [inline database/fake_data.sql:62]: Could some of these just have NULL values for region and constellation?

**Review by AlexOBarnes — APPROVED** [review]
Thanks for this test data very useful

**Comment by AlexOBarnes** [comment]
Maybe in future we could add comments to this file to explain what this is doing

**Comment by AlexOBarnes** [comment]
Also consider writing a readme for our test-data folder to explain how to use it

File: database/fake_data.sql [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+(13, 3, '2024-10-01 15:15:00', 240.1, 85.1),
+(14, 4, '2024-10-01 15:30:00', 300.1, 90.1),
+(15, 5, '2024-10-01 15:45:00', 330.1, 75.1),
+(16, 6, '2024-10-01 16:00:00', 15.1, 60.1),
+(17, 7, '2024-10-01 16:15:00', 60.1, 50.1),
+(18, 8, '2024-10-01 16:30:00', 105.1, 45.1),
+(19, 9, '2024-10-01 16:45:00', 150.1, 40.1),
+(20, 1, '2024-10-01 17:00:00', 195.1, 35.1),
+(21, 1, '2024-10-01 17:15:00', 240.1, 30.1),
+(22, 2, '2024-10-01 17:30:00', 285.1, 25.1),
```
Comment (CalebMathers) [inline database/fake_data.sql:253]: Region IDs shouldn't go above 12

File: database/fake_data.sql (hunk 4 above)
Comment (AlexOBarnes) [reply database/fake_data.sql:62]: I believe in the schema the constrains prevent nulls

File: database/fake_data.sql (hunk 3 above)
Comment (AlexOBarnes) [reply database/fake_data.sql:98]: Indubitably

File: database/fake_data.sql [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+('jacobwhite', '+12345678915', 'jacobwhite@example.com'),
+('oliviayoung', '+12345678916', 'oliviayoung@example.com'),
+('ethanturner', '+12345678917', 'ethanturner@example.com'),
+('gracehall', '+12345678918', 'gracehall@example.com'),
+('ryanallen', '+12345678919', 'ryanallen@example.com'),
+('amandacook', '+12345678920', 'amandacook@example.com');
+
+
+INSERT INTO subscriber_county_assignment (subscriber_id, county_id)
+VALUES
```
Comment (AlexOBarnes) [inline database/fake_data.sql:104]:
```suggestion
INSERT INTO subscriber_county_assignment (county_id, subscriber_id)
VALUES
```

Comment (AlexOBarnes) [reply database/fake_data.sql:104]: Found a bug!

File: database/fake_data.sql [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+('2024-10-01 07:55:00', '2024-10-01 20:10:00', 86),
+('2024-10-01 07:56:00', '2024-10-01 20:11:00', 87),
+('2024-10-01 07:57:00', '2024-10-01 20:12:00', 88),
+('2024-10-01 07:58:00', '2024-10-01 20:13:00', 89),
+('2024-10-01 07:59:00', '2024-10-01 20:14:00', 90),
+('2024-10-01 08:00:00', '2024-10-01 20:15:00', 91);
+
+
+INSERT INTO body_assignment (region_id, body_id, at, azimuth, altitude)
+VALUES
```
Comment (AlexOBarnes) [inline database/fake_data.sql:230]: Maybe include negative values for this section

**Review by AlexOBarnes — CHANGES_REQUESTED** [review]

**Review by AlexOBarnes — APPROVED** [review]
Thank for these what a splendid flabbergast!

File: database/fake_data.sql (hunk 4 above)
Comment (AlexOBarnes) [reply database/fake_data.sql:62]: Would be good to test the extended functionality of our ERD

Comment (MurasatoNaoya) [reply database/fake_data.sql:62]: Can confirm we can have nulls here, as the schema specifically allows it for region ID and constellation ID.

File: database/fake_data.sql (hunk 3 above)
Comment (MurasatoNaoya) [reply database/fake_data.sql:98]: Nice example of test emails. We can't test our SNS system works (actually sends messages) with this of course, but we can use them for ensuring our database is functioning as intended.

File: database/fake_data.sql (hunk 7 above)
Comment (MurasatoNaoya) [reply database/fake_data.sql:230]: Great mappings here, we can use this to test fetching astronomical positional data from the astronomyAPI for a given 'night period' (sunset). Of course, also allows us to test our database stores data as we intend.

**Review by MurasatoNaoya — APPROVED** [review]

**Comment by AlexOBarnes** [comment]
This closes #79 #77 #76 #74

**Pull request merged**